providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResourcesTest.java

```diff
@@ -0,0 +1,314 @@
+
+    public void testCleanupAvailabilitySetIfOrphaned() {
+        IdReference availabilitySetRef = IdReference.create("prefix/resourceGroups/myGroupName/myAvailabilitySet");
+        VirtualMachine vm = mockVmWithAvailabilitySet(availabilitySetRef);
+        AvailabilitySet availabilitySet = mockAvailabilitySet(ImmutableMap.of("jclouds", "myval"));
+
+        AzureComputeApi api = createMock(AzureComputeApi.class);
+        AvailabilitySetApi availabilitySetApi = createMock(AvailabilitySetApi.class);
+
+        expect(api.getAvailabilitySetApi("myGroupName")).andReturn(availabilitySetApi).atLeastOnce();
+
+        expect(availabilitySetApi.get("myAvailabilitySet")).andReturn(availabilitySet);
+        expect(availabilitySetApi.delete("myAvailabilitySet")).andReturn(URI.create("http://myDeletionUri"));
+
+        replay(api, availabilitySetApi);
+
+        // Do call
+        cleanupResources(api).cleanupAvailabilitySetIfOrphaned(vm);
+
+        // Verify deleted availability set
+        verify(api, availabilitySetApi);
+    }
+
+    public void testDeleteResourceGroupIfEmpty() {
+        AzureComputeApi api = createMock(AzureComputeApi.class);
+        ResourceGroupApi resourceGroupApi = createMock(ResourceGroupApi.class);
+
+        expect(api.getResourceGroupApi()).andReturn(resourceGroupApi).atLeastOnce();
+
+        expect(resourceGroupApi.resources("myGroupName")).andReturn(ImmutableList.<Resource>of());
+        expect(resourceGroupApi.delete("myGroupName")).andReturn(URI.create("http://myDeletionUri"));
+
+        replay(api, resourceGroupApi);
+
+        // Do call
+        cleanupResources(api).deleteResourceGroupIfEmpty("myGroupName");
+
+        // Verify deleted resource group
+        verify(api, resourceGroupApi);
+    }
+
+    public void testCleanupManagedDisks() {
+        OSDisk osDisk = mockOsDisk("prefix/resourceGroups/myGroupName/myOsDisk");
+        DataDisk dataDisk = mockDataDisk("prefix/resourceGroups/myGroupName/myDataDisk");
+        VirtualMachine vm = mockVmWithDisks(osDisk, ImmutableList.of(dataDisk));
+
+        AzureComputeApi api = createMock(AzureComputeApi.class);
+        DiskApi diskApi = createMock(DiskApi.class);
+        Disk disk = createMock(Disk.class);
+
+        expect(api.getDiskApi("myGroupName")).andReturn(diskApi).atLeastOnce();
+
+        expect(diskApi.delete("myOsDisk")).andReturn(URI.create("http://myDeletionUri"));
+        expect(diskApi.delete("myDataDisk")).andReturn(URI.create("http://myDeletionUri"));
+        // Simulate a delay in disk deletion
+        expect(diskApi.get("myDataDisk")).andReturn(disk).times(3);
+        expect(diskApi.get("myDataDisk")).andReturn(null).times(2); // An additional call is made when filtering the set
+        expect(diskApi.get("myOsDisk")).andReturn(null);
+
+        replay(api, diskApi);
+
+        // Do call
+        cleanupResources(api).cleanupManagedDisks(vm);
+
+        // Verify deleted resource group
+        verify(api, diskApi);
+    }
```

Comment:
Now that you're adding this (thanks!) add a method to verify that the virtual networks are deleted too.

Also add a test to verify that the resources that _don't_ have the jclouds tags (availability sets, networks, etc), are not deleted.